Code/GraphMol/Fingerprints/testFingerprintGenerators.cpp

```diff
@@ -2423,6 +2423,46 @@ void testBulkFP() {
   BOOST_LOG(rdErrorLog) << "  done" << std::endl;
 }
 
+void testAtomPairFPDifference() {
+  BOOST_LOG(rdErrorLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdErrorLog) << "    Test GitHub Issue #6958" << std::endl;
+  auto zinc36905137 = "Nc1nncc(-c2ccc(OC(F)F)cc2)n1"_smiles;
+  const unsigned int minLength = 1;
+  const unsigned int maxLength = 30;
+  const std::vector<std::uint32_t> fromAtoms{12, 13};
+  std::unique_ptr<SparseIntVect<std::int32_t>> fpFromFreeFunc(
+      RDKit::AtomPairs::getAtomPairFingerprint(*zinc36905137, minLength,
+                                               maxLength, &fromAtoms));
+  std::unique_ptr<FingerprintGenerator<std::uint32_t>> atomPairGenerator(
+      AtomPair::getAtomPairGenerator<std::uint32_t>(minLength, maxLength));
+  std::unique_ptr<SparseIntVect<std::uint32_t>> fpFromGenerator(
+      atomPairGenerator->getSparseCountFingerprint(*zinc36905137, &fromAtoms));
+  TEST_ASSERT(fpFromFreeFunc->size() == fpFromGenerator->size());
+  const auto &fpFromFreeFuncNonZero = fpFromFreeFunc->getNonzeroElements();
+  std::vector<std::pair<std::uint32_t, int>> fpFromFreeFuncAsVectOfPairs(
+      fpFromFreeFuncNonZero.begin(), fpFromFreeFuncNonZero.end());
+  std::sort(fpFromFreeFuncAsVectOfPairs.begin(),
+            fpFromFreeFuncAsVectOfPairs.end());
+  const auto &fpFromGeneratorNonZero = fpFromGenerator->getNonzeroElements();
+  std::vector<std::pair<std::uint32_t, int>> fpFromGeneratorAsVectOfPairs(
+      fpFromGeneratorNonZero.begin(), fpFromGeneratorNonZero.end());
+  std::sort(fpFromGeneratorAsVectOfPairs.begin(),
+            fpFromGeneratorAsVectOfPairs.end());
+  if (fpFromFreeFuncAsVectOfPairs != fpFromGeneratorAsVectOfPairs) {
+    for (const auto &fpFromFreeFuncPair : fpFromFreeFuncAsVectOfPairs) {
+      auto i = &fpFromFreeFuncPair - &fpFromFreeFuncAsVectOfPairs.front();
```

Comment:
I think this is kind of a gross pattern. What about:
```suggestion
    for (auto i = 0u; i < fpFromFreeFuncAsVectOfPairs.size(); ++i) {
      const auto &fpFromFreeFuncPair = fpFromFreeFuncAsVectOfPairs.at(i);
```

Reply:
@greglandrum _Gross_? That’s _art_ of pointer arithmetic. Anyway, I’ll commit the change. Only because it’s you ;-)